src/spec/test/DesignPatternsTest.groovy

```diff
@@ -24,7 +24,7 @@ class DesignPatternsTest extends CompilableTestSupport {
         shouldCompile '''
             // tag::abstract_factory_example1[]
             class TwoupMessages {
-                def welcome = 'Welcome to the twoup game, you start with $1000'
+                def welcome = 'Welcome to the twoup game, you start with \$1000'
```

Comment:
I actually tested the cede as it was as and it threw out some exception: 
```
org.codehaus.groovy.control.MultipleCompilationErrorsException: startup failed:
E:\Learn\groovy\DesignPatterns\AbstractFactoryPattern\test.groovy: 2: illegal string body character after dollar sign;
   solution: either escape a literal dollar sign "\$5" or bracket the value expression "${5}" @ line 2, column 64.
    two-up game, you start with $1000"
                                 ^

1 error
```

The version of groovy that I am using is 2.4.7